generators/base/logic/generator/locator.mjs

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2013-2022 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const locateGenerator = (generator, env, options) => {
+  let existingGenerator;
+  try {
+    existingGenerator = generator || env.requireNamespace(options.namespace).generator;
+  } catch (error) {
+    if (options.namespace) {
+      const split = options.namespace.split(':', 2);
+      existingGenerator = split.length === 1 ? split[0] : split[1];
+    }
+  }
+  return existingGenerator;
+};
```

Comment:
This is really confusing as a generator member, I think it's impossible to follow out of generator class context.

Reply:
I really don't want to have three different utils classes (`lib` folder, `utils.mjs, `logic`folder). which is why I added it there.
Also, it was originally part of `generator-base` so it does not harm more being here...

Reply:
rolledback